xds/src/main/java/io/grpc/xds/OrcaUtil.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.envoyproxy.udpa.data.orca.v1.OrcaLoadReport;
+import io.grpc.Attributes;
+import io.grpc.CallOptions;
+import io.grpc.ClientStreamTracer;
+import io.grpc.ClientStreamTracer.StreamInfo;
+import io.grpc.LoadBalancer;
+import io.grpc.Metadata;
+import io.grpc.protobuf.ProtoUtils;
+import io.grpc.util.ForwardingClientStreamTracer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility class that provides method for {@link LoadBalancer} to install listeners to receive
+ * backend cost metrics in the format of Open Request Cost Aggregation (ORCA).
+ */
+public final class OrcaUtil {
+
+  private static final ClientStreamTracer NOOP_CLIENT_STREAM_TRACER = new ClientStreamTracer() {};
+  private static final ClientStreamTracer.Factory NOOP_CLIENT_STREAM_TRACER_FACTORY =
```

Comment:
We had a couple of places using `NOOP_CLIENT_STREAM_TRACER_FACTORY ` (the other place I know is in `XdsLoadReportStore`). Should we make it as a static instance in `ClientStreamTracer.Factory`?

Reply:
No. It's trivial to implement and it doesn't have much value on the public API.